ipn/ipnlocal/local.go

```diff
@@ -4856,6 +4857,16 @@ func (b *LocalBackend) applyPrefsToHostinfoLocked(hi *tailcfg.Hostinfo, prefs ip
 	}
 	hi.SSH_HostKeys = sshHostKeys
 
+	services := b.vipServicesFromPrefs(prefs)
+	if len(services) > 0 {
+		var buf bytes.Buffer
+		json.NewEncoder(&buf).Encode(services)
```

Comment:
Or just:

```go
buf, _ := json.Marshal(services)
hi.ServicesHash = fmt.Sprintf("%02x", sha256.Sum256(buf))
```

Or:

```go
    hi.ServicesHash = deephash.Hash(&services).String()
```

Reply:
Oh, wait. From deephash docs:

> Users should not store the hash on disk or send it over the network.

We will absolutely be sending this over the network. :sweat_smile: I'll put sha256 back in

Reply:
Sure, sha256 is fine. Deephash is just saying it's subject to change, algorithm wise. But in our case, if it changes that just means control would c2n to us and find no changes.

But I guess it's safer to be more stable.